cmd/cmrel/cmd/gcb_bootstrap_pgp.go

```diff
@@ -0,0 +1,128 @@
+	gcbBootstrapPGPCommand         = bootstrapPGPCommand
+	gcbBootstrapPGPDescription     = bootstrapPGPDescription
+	gcbBootstrapPGPLongDescription = `The bootstrap-pgp command uses a specified KMS key in GCP to create a PGP identity
+which can be used for code signing. The public identity can then be
+distributed for use in signature verification, while the KMS key itself can be
+used to sign cert-manager artifacts.
+
+The public key is written to stdout in "armored" format.
+
+The raw PEM-encoded public key is also written to stdout.
+
+This is the internal version of the 'bootstrap-pgp' target. It is intended to be run by
+a Google Cloud Build started via the 'bootstrap-pgp' sub-command.`
+)
+
+var gcbBootstrapPGPExample = fmt.Sprintf(`To create a public PGP identity for a key:
+
+%s gcb %s --key "projects/<PROJECT_NAME>/locations/<LOCATION>/keyRings/<KEYRING_NAME>/cryptoKeys/<KEY_NAME>/cryptoKeyVersions/<KEY_VERSION>"`, rootCommand, gcbBootstrapPGPCommand)
+
+type gcbBootstrapPGPOptions struct {
+	// Key is the full name of the GCP KMS key to be used, e.g.
+	// projects/<PROJECT_NAME>/locations/<LOCATION>/keyRings/<KEYRING_NAME>/cryptoKeys/<KEY_NAME>/cryptoKeyVersions/<KEY_VERSION>
+	Key string
+}
+
+func (o *gcbBootstrapPGPOptions) AddFlags(fs *flag.FlagSet, markRequired func(string)) {
+	fs.StringVar(&o.Key, "key", "", "Full name of the GCP KMS key to use for bootstrapping")
+	markRequired("key")
+}
+
+func (o *gcbBootstrapPGPOptions) print() {
+	log.Printf("bootstrap-pgp options:")
+	log.Printf("         Key: %q", o.Key)
+}
+
+func gcbBootstrapPGPCmd(rootOpts *rootOptions) *cobra.Command {
+	o := &gcbBootstrapPGPOptions{}
+	cmd := &cobra.Command{
+		Use:          gcbBootstrapPGPCommand,
+		Short:        gcbBootstrapPGPDescription,
+		Long:         gcbBootstrapPGPLongDescription,
+		Example:      gcbBootstrapPGPExample,
+		SilenceUsage: true,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			o.print()
+			log.Printf("---")
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runGCBBootstrapPGP(rootOpts, o)
+		},
+	}
+
+	o.AddFlags(cmd.Flags(), mustMarkRequired(cmd.MarkFlagRequired))
+
+	return cmd
+}
+
+func getPEMPubkey(ctx context.Context, key string) (string, error) {
+	oauthClient, err := google.DefaultClient(ctx, cloudkms.CloudPlatformScope)
+	if err != nil {
+		return "", fmt.Errorf("could not create GCP OAuth2 client: %w", err)
+	}
+
+	svc, err := cloudkms.New(oauthClient)
+	if err != nil {
+		return "", fmt.Errorf("could not create GCP KMS client: %w", err)
+	}
+
+	res, err := svc.Projects.Locations.KeyRings.CryptoKeys.CryptoKeyVersions.GetPublicKey(key).Do()
+	if err != nil {
+		return "", fmt.Errorf("could not get public key from Google Cloud KMS API: %w", err)
+	}
+
+	return res.Pem, nil
+}
+
+func runGCBBootstrapPGP(rootOpts *rootOptions, o *gcbBootstrapPGPOptions) error {
+	ctx := context.Background()
+
+	armoredKey, err := sign.BootstrapPGPFromGCP(ctx, o.Key)
```

Comment:
This actually is idempotent; it'll generate a new identity which will only differ in its creation time, and the old one will still be perfectly valid. It's safe for us to run this as much as we like, although there's no reason to run it multiple times since the first identity generated for a key will work just fine.

The identity is kinda like the public part of an X.509 cert.

It could be made to store the generated identity for a named KMS key, but I'm not really sure what that gains us beyond saving a `Sign` API call for the key?

Reply:
> It could be made to store the generated identity for a named KMS key, but I'm not really sure what that gains us beyond saving a Sign API call for the key?

Agree that that doesn't make sense. I was only concerned that re-running the bootstrap would result in the generation of new private key so the existing distributed public keys wouldn't work anymore, but I've misunderstood this bit